k9mail-library/src/main/java/com/fsck/k9/mail/internet/MimeBodyPart.java

```diff
@@ -36,6 +37,11 @@ public MimeBodyPart(Body body, String mimeType) throws MessagingException {
         MimeMessageHelper.setBody(this, body);
     }
 
+    MimeBodyPart(MimeHeader header, Body body)  throws MessagingException {
```

Comment:
If you make the other constructors call this one, you can keep `mHeader` as `final`